controller/src/controller/db/client.py

```diff
@@ -0,0 +1,688 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import List, Optional, Type, Union
+
+import genai_factory.schemas as api_models
+
+
+class Client(ABC):
+    """
+    Base class for all DB clients
+    In order to implement a new DB client, inherit from this class and implement all the methods.
+    """
+
+    @abstractmethod
+    def __init__(self):
+        pass
```

Comment:
I don't see the point in forcing init method. You only force him to have an init without any arguments as inputs, which is not true as you can pass db configurations via the controller config and send them to the client in his init.

Reply:
So the db configuration should be in separate variable in the config?

Reply:
well, yeah I mean every DB will probably have its own stuff. So we can put it under "db" field and everything there will be passed as `MyChosenClient(**config["db"])`